erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -213,6 +214,19 @@ def check_duplicate_entry_for_production_order(self):
 				frappe.throw(_("Stock Entries already created for Production Order ")
 					+ self.production_order + ":" + ", ".join(other_ste), DuplicateEntryForProductionOrderError)
 
+	def set_incoming_rate(self):
+		for d in self.items:
+			if d.s_warehouse:
+				args = self.get_args_for_incoming_rate(d)
+				d.basic_rate = get_incoming_rate(args)
+			elif d.t_warehouse and not d.basic_rate:
+				warehouse = d.s_warehouse or d.t_warehouse
+				d.basic_rate = get_valuation_rate(d.item_code, warehouse,
+					self.doctype, d.name, d.allow_zero_valuation_rate,
+					currency=erpnext.get_company_currency(self.company))
+			elif d.allow_zero_valuation_rate and not d.s_warehouse:
```

Comment:
This `elif` condition should be in 2nd position. If there is s_warehouse this elif will not be executed